crypto/src/main/java/org/hyperledger/besu/crypto/NodeKey.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.crypto;
+
+import org.hyperledger.besu.crypto.SECP256K1.PublicKey;
+import org.hyperledger.besu.crypto.SECP256K1.Signature;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+
+public interface NodeKey {
+
+  Signature sign(Bytes32 dataHash);
+
+  PublicKey getPublicKey();
+
+  Bytes32 calculateKeyAgreement(final PublicKey partyKey);
+
+  Bytes calculateECIESAgreement(final SECP256K1.PublicKey ephPubKey);
```

Comment:
You could have the method type consistent with above method on the interface?
```suggestion
  Bytes calculateECIESAgreement(final PublicKey ephPubKey);
```